tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_reducers_test.ts

```diff
@@ -870,6 +870,75 @@ describe('Debugger reducers', () => {
     expect(nextState.executions.scrollBeginIndex).toBe(1450);
   });
 
+  for (const scrollIndex of [0, 1, 20, 50]) {
+    it(`executionScrollToIndex sets correct scrollBeginIndex ${scrollIndex}`, () => {
+      const scrollBeginIndex = 0;
+      const displayCount = 50;
+      const opTypes = new Array<string>(100);
+      opTypes.fill('FooOp');
+      const state = createDebuggerStateWithLoadedExecutionDigests(
+        scrollBeginIndex,
+        displayCount,
+        opTypes
+      );
+      const nextState = reducers(
+        state,
+        actions.executionScrollToIndex({index: scrollIndex})
+      );
+      expect(nextState.executions.scrollBeginIndex).toBe(scrollIndex);
+    });
+  }
+
+  for (const scrollIndex of [-1, 0.5, 51, 100]) {
+    it(
+      `Invalid executionScrollToIndex (${scrollIndex}) does not change scrollBeginIdnex:` +
+        `displayCount < numExecutions`,
+      () => {
+        const originalScrollBeginIndex = 3;
+        const displayCount = 50;
+        const opTypes = new Array<string>(100);
+        opTypes.fill('FooOp');
+        const state = createDebuggerStateWithLoadedExecutionDigests(
+          originalScrollBeginIndex,
+          displayCount,
+          opTypes
+        );
+        const nextState = reducers(
+          state,
+          actions.executionScrollToIndex({index: scrollIndex})
+        );
+        expect(nextState.executions.scrollBeginIndex).toBe(
+          originalScrollBeginIndex
+        );
+      }
+    );
+  }
+
+  for (const scrollIndex of [-1, 0.5, 1, 2, 20]) {
```

Comment:
I don't think I am understanding the test by itself. I get -1 and 0.5 but why are 1 and 2 problematic? What about 20?

Reply:
Here, displayCount is 50 and there are only 20 execution digests (< 50). In this case, the scroll bar is hidden in the TimelineComponent and scrolling is disabled (i.e., the only valid scrolling begin index is 0). I added a comment here to clarify that.